### ISSUE TITLE
RPC: keygen run error: no login-UI was available after select key on signup

### ISSUE THREAD

**Issue by gabriel** [keybase/go#192]
Getting the error: `keygen run error: no login-UI was available` on signup after selecting a PGP key.

```
2015-02-17 18:33:29.527 Keybase[32412:8185769] -[KBRPClient sendRequestWithMethod:params:completion:]:138: Sent request: keybase.1.signup.signup({
    deviceName = Test;
    email = "[FILTERED]";
    inviteCode = "[FILTERED]";
    passphrase = "[FILTERED]";
    username = gbrl45;
})
2015-02-17 18:33:35.462 Keybase[32412:8185769] __18-[KBRPClient open]_block_invoke:48: Received request: keybase.1.gpgUi.selectKeyAndPushOption({
....
})
2015-02-17 18:33:40.227 Keybase[32412:8185769] __18-[KBRPClient open]_block_invoke:48: Received request: keybase.1.secretUi.getSecret({
    pinentry =     {
        cancel = "";
        desc = "Please enter the passphrase to unlock the secret key for:\nuser: Gabriel Test <gabrielh+gbrl@gmail.com>\n4096-bit RSA key, ID E1A69836A122EBA8, created 2015-02-17\n\nReason: Import of key into keybase keyring";
        err = "";
        ok = "";
        prompt = "Your key passphrase";
    };
})
2015-02-17 18:33:40.234 Keybase[32412:8185769] __45-[AppDelegate applicationDidFinishLaunching:]_block_invoke:46: Password prompt: (
        {
        pinentry =         {
            cancel = "";
            desc = "Please enter the passphrase to unlock the secret key for:\nuser: Gabriel Test <gabrielh+gbrl@gmail.com>\n4096-bit RSA key, ID E1A69836A122EBA8, created 2015-02-17\n\nReason: Import of key into keybase keyring";
            err = "";
            ok = "";
            prompt = "Your key passphrase";
        };
    }
)
2015-02-17 18:33:45.763 Keybase[32412:8185769] __54-[KBRPClient sendRequestWithMethod:params:completion:]_block_invoke:130: Error: Error Domain=MPMessagePack Code=218 "keygen run error: no login-UI was available" UserInfo=0x610000466ac0 {NSLocalizedDescription=keygen run error: no login-UI was available}
2015-02-17 18:33:45.763 Keybase[32412:8185769] __54-[KBRPClient sendRequestWithMethod:params:completion:]_block_invoke:131: Result: {
    passphraseOk = 0;
    postOk = 0;
    writeOk = 0;
}
```

@patrickxb this is probably you right?

**Comment by maxtaco** [comment]
Still buggy for pushing a private key to server; also, lots of password prompting...

**Comment by patrickxb** [comment]
Hmmm.  I can look into this and #193.

**Comment by patrickxb** [comment]
This should be fixed with 44bf3c79a6edb7046d826d457b1706fe3c3c0614
Reopen if not.

I haven't touched #193 yet, but the missing ui bug should be fixed.
